include/nighthawk/client/factories.h

```diff
@@ -37,9 +37,9 @@ class SequencerFactory {
 public:
   virtual ~SequencerFactory() = default;
   virtual SequencerPtr create(Envoy::TimeSource& time_source, Envoy::Event::Dispatcher& dispatcher,
-                              Envoy::MonotonicTime start_time, BenchmarkClient& benchmark_client,
-                              TerminationPredicate& termination_predicate,
-                              Envoy::Stats::Scope& scope) const PURE;
+                              BenchmarkClient& benchmark_client,
+                              TerminationPredicatePtr&& termination_predicate,
+                              Envoy::Stats::Scope& scope, const bool warmup) const PURE;
```

Comment:
Similar to our discussion in one of the previous PRs, how about we use an enum instead of the boolean flag argument? Looks like this boolean is used in a couple of interfaces that plumb the phases through. We might be able to reuse the enum in multiple locations.

This however makes me wonder how the end result is going to look like. If we go down the path of dynamically configured phases, what are we going to replace the boolean here with? Do we need a class carrying information about phases?

With that said - is this something we should leave to the next PR? Or would it be more beneficial to come up with a good structure now?

Reply:
I'm happy to iterate here. My goal with this is to land a decent chunk of the changes required to implement phases as a no-op, as an intermediate step to make review more fun. 
Later on we'll use them generically, we can also iterate on this in here in this PR here if you prefer. Or we can go for shipping this, and iterate on a second step in another one.

Reply:
I eliminated the warmup bool arg, to minimize changes that won't have future value as per earlier discussion.